### ISSUE TITLE
Additional fields needed in ConferenceRoom data model

### ISSUE THREAD

**Issue by agis-jupiter** [ant-media/conference-call-application#38]
The ConferenceRoom model used in the REST API only has 6 fields.
These fields are roomId, startDate, endDate, roomStreamList, mode, and originAdress.

For the Broadcast object there are many unused fields that can be used however needed.
For the ConferenceRoom object, that is not currently really possible.

The object missing some fields that I believe would be useful for many people.
It would be very helpful to have an additional field for something like a room Topic, Description, or Purpose of the conference call.
It would also be very helpful to have an additional field for a list of names of people that are invited/expected to join conference.  (This is different than the people who have already joined and and are streaming.)

Since there are so few fields to use and a conference can be a pretty complex thing, it might also be useful to have so extra fields thrown in for use by people implementing their own conference pages.
Something like fields Extra1, Extra2, Extra3 for example. Just simply some extra fields people can use for more information about the conference.

**Comment by mekya** [comment]
Hey @agis-jupiter ,

Thank you for the issue. I personally thinking of removing the ConferenceRoom Model and use Broadcast in every models. Because in new conferencing UI, multitrack playback is used. 

Additionally, we can add some metadata object so that you can add a new kind of data to the structure. 

Is this something OK for you?

Regards
Oguz

**Comment by agis-jupiter** [comment]
Yes, adding a metadata field for custom data would be very useful.  I do not understand what you mean about "removing the ConferenceRoom Model and use Broadcast". So when we would do a REST call to get the conference rooms (i.e. [/v2/broadcasts/conference-rooms/list/{offset}/{size}]) we would then receive a list of "Broadcasts" objects rather than "ConferenceRoom" objects?

**Comment by mekya** [comment]
Yes and No :)

>So when we would do a REST call to get the conference rooms (i.e. [/v2/broadcasts/conference-rooms/list/{offset}/{size}]) we would then receive a list of "Broadcasts" objects rather than "ConferenceRoom" objects?

We'll make the REST API backward compatible.  "Broadcast" objects will be converted to "ConferenceRoom" objects for the sake of compatibility. You think you receive "ConferenceRoom" objects. On the other hand, they will be "Broadcast" objects.

I hope it's OK for you. If not, please let me know.

**Comment by agis-jupiter** [comment]
Before suggesting we move forward, I just want to make sure we are not losing any ability.
It sounds like we wont lose anything since you are trying to maintain backwards compatibility, but I just wanted to be sure.

It sounds like when we request to get the Conferences that have been created, we will receive a list of ConferenceRooms (that are really Broadcasts behind the scenes.)
I think that is fine as long as we do not lost any data that we need.


Will the JSON object returned have the old "ConferenceRoom" fields and also have the "Broadcast" fields added to it now?
In our case, I need to make sure that we when we make the REST request /v2/broadcasts/conference-rooms/list, we know 
a. any information needed to join the conference (e.g. roomId, etc...)
b. when the conferences start
c. when the conferences end
d. and any data we add to the new custom metadata object/field for each conference.


Also will there be a size or character limit to the new custom metadata object?

**Comment by agis-jupiter** [comment]
Is it possible to get a rough time frame of when this proposed plan would be implemented?
This is a pretty critical piece for our next software release to include conferencing.

Also, in the mean time, can we use one of the existing fields in the ConferenceRoom model to store our metadata?
(i.e. startDate, endDate, roomStreamList, mode, originAdress)
Are any of these unused by the server?
Specifically, would it cause a problem for the server if we store our metadata in any of these fields?

**Comment by burak-58** [comment]
Hi @agis-jupiter ,
**Information**
We may deprecate ConferenceRoom model in future releases. In track based conference solution like this project, a [Broadcast](https://github.com/ant-media/Ant-Media-Server/blob/master/src/main/java/io/antmedia/datastore/db/types/Broadcast.java) object is created and used as main track. All the participants' streams are the sub tracks of this main track. The stream id of that Broadcast is same with the room id. Normally it is created automatically when the fitst participant joins.
You may check [this](https://antmedia.io/reveal-the-secrets-of-3-types-of-video-conference-solutions/) to understand the different types of conference solutions for AMS.

**Solution to your issue:**
You can create this broadcast with [this REST method](https://antmedia.io/rest/#/BroadcastRestService/createBroadcast). So you can set the metaData field as you want. We want to use this Broadcast object instead of ConferenceRoom in future releases. So this may be a better solution for you.

Here is a sample curl command to create such a Broadcast with metaData:

```
curl -X POST -H 'Content-Type: application/json' \
	http://localhost:5080/WebRTCAppEE/rest/v2/broadcasts/create \
      -d '{ "streamId":"room1","metaData":"this is my custom data"}' 
```

**Comment by burak-58** [comment]
Closing due to inactivity.
